cli/command/trust/signer_remove.go

```diff
@@ -0,0 +1,137 @@
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Do not prompt for confirmation before removing the most recent signer")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, options signerRemoveOptions) error {
+	var errImages []string
+	for _, image := range options.images {
+		if err := removeSingleSigner(cli, image, options.signer, options.forceYes); err != nil {
+			fmt.Fprintln(cli.Out(), err.Error())
+			errImages = append(errImages, image)
+		}
+	}
+	if len(errImages) > 0 {
+		return fmt.Errorf("Error removing signer from: %s", strings.Join(errImages, ", "))
+	}
+	return nil
+}
+
+func isLastSignerForReleases(roleWithSig data.Role, allRoles []client.RoleWithSignatures) (bool, error) {
+	var releasesRoleWithSigs client.RoleWithSignatures
+	for _, role := range allRoles {
+		if role.Name == releasesRoleTUFName {
+			releasesRoleWithSigs = role
+			break
+		}
+	}
+	counter := len(releasesRoleWithSigs.Signatures)
+	if counter == 0 {
+		return false, fmt.Errorf("All signed tags are currently revoked, use docker trust sign to fix")
+	}
+	for _, signature := range releasesRoleWithSigs.Signatures {
+		for _, key := range roleWithSig.KeyIDs {
+			if signature.KeyID == key {
+				counter--
+			}
+		}
+	}
+	return counter < releasesRoleWithSigs.Threshold, nil
+}
+
+func removeSingleSigner(cli command.Cli, imageName, signerName string, forceYes bool) error {
+	fmt.Fprintf(cli.Out(), "\nRemoving signer \"%s\" from %s...\n", signerName, imageName)
+
+	ctx := context.Background()
+	imgRefAndAuth, err := trust.GetImageReferencesAndAuth(ctx, image.AuthResolver(cli), imageName)
+	if err != nil {
+		return err
+	}
+
+	signerDelegation := data.RoleName("targets/" + signerName)
+	if signerDelegation == releasesRoleTUFName {
+		return fmt.Errorf("releases is a reserved keyword and cannot be removed")
+	}
+	notaryRepo, err := cli.NotaryClient(imgRefAndAuth, trust.ActionsPushAndPull)
+	if err != nil {
+		return trust.NotaryError(imgRefAndAuth.Reference().Name(), err)
+	}
+	delegationRoles, err := notaryRepo.GetDelegationRoles()
+	if err != nil {
+		return fmt.Errorf("Error retrieving signers for %s", imageName)
+	}
+	var role data.Role
+	for _, delRole := range delegationRoles {
+		if delRole.Name == signerDelegation {
+			role = delRole
+			break
+		}
+	}
+	if role.Name == "" {
+		return fmt.Errorf("No signer %s for image %s", signerName, imageName)
+	}
+	allRoles, err := notaryRepo.ListRoles()
+	if err != nil {
+		return err
+	}
+	if ok, err := isLastSignerForReleases(role, allRoles); ok && !forceYes {
+		removeSigner := command.PromptForConfirmation(os.Stdin, cli.Out(), fmt.Sprintf("The signer \"%s\" signed the last released version of %s. "+
+			"Removing this signer will make %s unpullable. "+
+			"Are you sure you want to continue?",
+			signerName, imageName, imageName,
+		))
+
+		if !removeSigner {
+			fmt.Fprintf(cli.Out(), "\nAborting action.\n")
+			return nil
+		}
+	} else if err != nil {
+		fmt.Fprintln(cli.Out(), err.Error())
```

Comment:
Shouldn't this return the error here to exit early?

If not I guess just `s/cli.Out/cli.Err/`